### PR TITLE
doc: Add nRF53 to Thread architecture user guide

### PR DESCRIPTION
**Pull request by MarekPorwisz**
Thread on nRF53 series devices architecture is different than on nRF52
and it needs to be reflected in docs as well.

Signed-off-by: Marek Porwisz <marek.porwisz@nordicsemi.no>

### CODE REVIEW

**Review by pszkotak — APPROVED** [review]

**Review by edmont — APPROVED** [review]

**Review by hubertmis — APPROVED** [review]

File: doc/nrf/images/thread_platform_design_nRF53.svg [hunk 1]
```diff
@@ -0,0 +1,424 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ru-fu) [inline doc/nrf/images/thread_platform_design_nRF53.svg:1]: Please add the Visio source file as well (for both images).

File: doc/nrf/ug_thread_architectures.rst [hunk 2]
```diff
@@ -51,14 +51,19 @@ This design has the following advantages:
 
 It also has the following disadvantages:
 
-* For some uses, the nRF528xx MCU can be too slow. For example, when the application would like to do complex data processing.
+* For some uses, the nRF528xx and nRF53xx MCUs can be too slow. For example, when the application would like to do complex data processing.
```
Comment (ru-fu) [inline doc/nrf/ug_thread_architectures.rst:54]:
```suggestion
* For some use cases, the nRF52 Series and nRF53 Series MCUs can be too slow (for example, when the application does complex data processing).
```

File: doc/nrf/ug_thread_architectures.rst [hunk 3]
```diff
@@ -51,14 +51,19 @@ This design has the following advantages:
...
 * The application and the network share Flash and RAM space, which can limit the application functionality.
 * Dual-bank DFU or an external flash is needed to update the firmware.
 
 .. figure:: /images/thread_platform_design_soc.svg
-   :alt: Thread-only architecture
+   :alt: Thread-only architecture (nRF52)
 
-   Thread-only architecture
+   Thread-only architecture on nRF52 series devices
```
Comment (ru-fu) [inline doc/nrf/ug_thread_architectures.rst:61]:
```suggestion
   Thread-only architecture on nRF52 Series devices
```

File: doc/nrf/ug_thread_architectures.rst [hunk 4]
```diff
@@ -51,14 +51,19 @@ This design has the following advantages:
...
+
+.. figure:: /images/thread_platform_design_nRF53.svg
+   :alt: Thread-only architecture (nRF53)
+
+   Thread-only architecture on nRF53 series devices
```
Comment (ru-fu) [inline doc/nrf/ug_thread_architectures.rst:66]:
```suggestion
   Thread-only architecture on nRF53 Series devices
```

File: doc/nrf/ug_thread_architectures.rst [hunk 5]
```diff
@@ -81,7 +86,7 @@ This platform design is suitable for the following development kits:
 Single-chip, multiprotocol (SoC)
 ================================
 
-With nRF52840 and nRF52833 supporting multiple wireless technologies, including IEEE 802.15.4 and Bluetooth Low Energy (Bluetooth LE), the application layer and OpenThread still run on the same processor.
+With nRF52 and nRF53 series devices supporting multiple wireless technologies, including IEEE 802.15.4 and Bluetooth Low Energy (Bluetooth LE), the application layer and OpenThread still run on the same processor.
```
Comment (ru-fu) [inline doc/nrf/ug_thread_architectures.rst:89]:
```suggestion
With nRF52 and nRF53 Series devices supporting multiple wireless technologies, including IEEE 802.15.4 and Bluetooth Low Energy (Bluetooth LE), the application layer and OpenThread still run on the same processor.
```

File: doc/nrf/ug_thread_architectures.rst [hunk 6]
```diff
@@ -95,9 +100,14 @@ It also has the following disadvantages:
 * Bluetooth LE activity can degrade the connectivity on Thread if not implemented with efficiency in mind.
 
 .. figure:: /images/thread_platform_design_multi.svg
-   :alt: Multiprotocol Thread and Bluetooth LE architecture
+   :alt: Multiprotocol Thread and Bluetooth LE architecture (nRF52)
+
+   Multiprotocol Thread and Bluetooth LE architecture on nRF52 series devices
```
Comment (ru-fu) [inline doc/nrf/ug_thread_architectures.rst:105]:
```suggestion
   Multiprotocol Thread and Bluetooth LE architecture on nRF52 Series devices
```

File: doc/nrf/ug_thread_architectures.rst [hunk 7]
```diff
@@ -95,9 +100,14 @@ It also has the following disadvantages:
...
+
+.. figure:: /images/thread_platform_design_nRF53_multi.svg
+   :alt: Multiprotocol Thread and Bluetooth LE architecture (nRF53)
 
-   Multiprotocol Thread and Bluetooth LE architecture
+   Multiprotocol Thread and Bluetooth LE architecture on nRF53 series devices
```
Comment (ru-fu) [inline doc/nrf/ug_thread_architectures.rst:110]:
```suggestion
   Multiprotocol Thread and Bluetooth LE architecture on nRF53 Series devices
```

**Review by ru-fu — CHANGES_REQUESTED** [review]

File: doc/nrf/images/thread_platform_design_nRF53.svg (hunk 1 above)
Comment (MarekPorwisz) [reply doc/nrf/images/thread_platform_design_nRF53.svg:1]: I do not have Visio. I've added Libre Draw sources that can be used to export to any other format accepted by visio.

**Review by ru-fu — APPROVED** [review]

**Pull request merged**